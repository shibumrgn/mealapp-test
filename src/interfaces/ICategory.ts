export interface ICategory {
    strCategory: string;
}
