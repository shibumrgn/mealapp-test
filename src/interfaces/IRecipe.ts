export interface IRecipe {
    idMeal: string;
    strMeal: string;
    strDrinkAlternate: string;
    strCategory: string;
    strArea: string;
    strInstructions: string;
    strMealThumb: string;
    strTags: string;
    strYoutube: string;
    ingredients: {
        name: string,
        amount: string
    }[]
}