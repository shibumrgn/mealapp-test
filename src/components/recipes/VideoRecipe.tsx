import {View, Text, Image, StyleSheet} from 'react-native';
import React, {useMemo} from 'react';
import {IRecipe} from '../../interfaces/IRecipe';
import {Brand} from '../../constants/Brand';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {useDispatch, useSelector} from 'react-redux';
import {RecipeActions} from '../../redux/reducers/recipeReducer';

interface IProps {
    recipe: IRecipe;
    width?: number | string;
}

const VideoRecipe = (props: IProps) => {
    const dispatch = useDispatch();
    const savedRecipes: IRecipe[] = useSelector(
        (state: any) => state.recipe?.savedRecipes,
    );

    const _saveRecipe = () => {
        dispatch(RecipeActions.saveRecipe(props.recipe));
    };

    const _removeRecipe = () => {
        dispatch(RecipeActions.removeRecipe(props.recipe));
    };

    const _isBookMarked = useMemo(
        () =>
            savedRecipes?.find(
                (sr: IRecipe) => props?.recipe?.idMeal === sr?.idMeal,
            ),
        [props.recipe, savedRecipes],
    );

    const styles = StyleSheet.create({
        parent: {
            width: props?.width || '100%',
            marginEnd: 10,
            marginBottom: 20,
        },
        image: {
            width: '100%',
            height: 200,
            marginBottom: 10,
            position: 'relative',
            borderRadius: 10,
            resizeMode: 'contain',
        },
        profile: {
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
        },
        profileImage: {
            borderRadius: 32,
            marginEnd: 5,
            width: 32,
            height: 32,
        },
        title: {...Brand.subText, marginBottom: 10},
    });

    return (
        <View style={styles.parent}>
            <Image
                source={{
                    uri: props?.recipe?.strMealThumb,
                }}
                style={styles.image}
            />
            <View
                style={{
                    position: 'absolute',
                    top: 0,
                    right: 0,
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    backgroundColor: 'white',
                    margin: 10,
                    borderRadius: 32,
                    height: 32,
                    width: 32,
                }}>
                <Ionicons
                    name="bookmark-outline"
                    size={18}
                    color={_isBookMarked ? Brand.accentColor : Brand.black}
                    onPress={!_isBookMarked ? _saveRecipe : _removeRecipe}
                />
            </View>
            <View>
                <Text style={styles.title}>{props.recipe?.strMeal}</Text>
                <View style={styles.profile}>
                    <Image
                        source={{
                            uri: 'https://picsum.photos/200/300',
                        }}
                        style={styles.profileImage}
                    />
                    <Text style={Brand.tiny}>By Shibu Murugan</Text>
                </View>
            </View>
        </View>
    );
};

export default VideoRecipe;
