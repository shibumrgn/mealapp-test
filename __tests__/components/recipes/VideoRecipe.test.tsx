
import 'react-native';
import React from 'react'; 
import renderer from 'react-test-renderer';
import SampleRecipes from '../../SampleRecipes.json'
import { IRecipe } from '../../../src/interfaces/IRecipe';
import VideoRecipe from '../../../src/components/recipes/VideoRecipe';
 
 it('renders correctly', () => {
   const render = renderer.create(<VideoRecipe recipe={SampleRecipes[0] as any as IRecipe}/>)?.toJSON();
   expect(render).toMatchSnapshot();
 });
 