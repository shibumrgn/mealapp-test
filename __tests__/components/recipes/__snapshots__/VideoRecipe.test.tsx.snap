// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginEnd": 10,
      "width": 562.5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://www.themealdb.com/images/media/meals/ustsqw1468250014.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 200,
        "marginBottom": 10,
        "position": "relative",
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Spicy Arrabiata Penne
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/200/300",
          }
        }
        style={
          Object {
            "borderRadius": 32,
            "height": 32,
            "marginEnd": 5,
            "width": 32,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        By Shibu Murugan
      </Text>
    </View>
  </View>
</View>
`;
